test/test_sentry_processor.py

```diff
@@ -77,6 +77,25 @@ def test_sentry_log_failure(mocker, level):
     )
 
 
+@pytest.mark.parametrize("level", ["error", "critical"])
+def test_sentry_log_failure_exc_info_true(mocker, level):
+    """Make sure sentry_sdk.utils.exc_info_from_error doesn't raise ValueError
+    Because it can't introspect exc_info.
+    Bug triggered when logger.error(..., exc_info=True) or logger.exception(...)
+    are used.
+    """
+    m_capture_event = mocker.patch("structlog_sentry.capture_event")
+
+    event_data = {"level": level, "event": level + " message", "exc_info": True}
+    processor = SentryProcessor(level=getattr(logging, level.upper()))
+    try:
+        1 / 0
+    except ZeroDivisionError:
+        processor(None, None, event_data)
+
+    assert m_capture_event.call_count == 1
```

Comment:
I would change the assertion to something like `m_capture_event.assert_called_once_with(...what_it_was_called_with...)`.
What do you think?

Reply:
Too complex as the traceback contains absolute path depending on the execution environment. What matters is that no `ValueError` is raised.
Since most of the test suite is mocking usage of sentry-sdk, the test suite is quite weak in this regard, because we were unable to catch a bug there. This added test case as is,  covers a lot because we don't mock `sentry-sdk`. The absence of error during test execution, is also valid, assertions are not always necessary.

You can give it a try and may be point me to some directions ?

Reply:
What about this? It's not the best thing, but at least we are testing that `ZeroDivisionError` was passed to `capture_event`, ...

```suggestion
    m_capture_event.assert_called_once()
    _, kwargs = m_capture_event.call_args
    assert kwargs["hint"]["exc_info"][0] == ZeroDivisionError
```

Reply:
Thanks for this @paveldedik

Reply:
No problem, unfortunately `assert_called_once` is not supported by python 3.5, my mistake, sorry for that, let's change it back to

```suggestion
    assert m_capture_event.call_count == 1
```

Reply:
Will add new suggestion in another review.